internal/provider/telemetry_resource.go

```diff
@@ -160,7 +174,22 @@ func (r *TelemetryResource) Read(ctx context.Context, req resource.ReadRequest,
 	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
 
 	if resp.Diagnostics.HasError() {
-		return
+		resp.Diagnostics.Append()
+	}
+
+	data.ModuleSource = basetypes.NewStringNull()
```

Comment:
Duplicate code. Would it make sense to extract a function for this?

Reply:
Good idea - added `data = updateModuleSourceAndVersion(data)`